### PR TITLE
Prop for empty search fix

### PR DESCRIPTION
**Pull request by ashishzingh**
This exposes a prop named onEmptySearch using which we can render a view when the search results are empty.

### CODE REVIEW

**Comment by ashishzingh** [comment]
This does what we wanted to.
I have a few doubts.
1) what should be the prop type of the JSX element to be passed?
2) whenever the subtree is not empty i am getting a warning listed below 
d3
the props for the search were
d4

File: src/components/MuiTreeView/index.jsx [hunk 1]
```diff
@@ -265,6 +276,10 @@ MuiTreeView.propTypes = {
   listItemProps: object,
   /** If true, search is case sensitive. Defaults to false. */
   caseSensitiveSearch: bool,
+  /** Property applied to render data when search result is empty.
+   * Defaults to null
+   * */
+  onEmptySearch: any,
```
Comment (helfi92) [inline src/components/MuiTreeView/index.jsx:282]: I think this could be `node` rather than `any`.

File: src/components/MuiTreeView/index.jsx [hunk 2]
```diff
@@ -265,6 +276,10 @@ MuiTreeView.propTypes = {
   listItemProps: object,
   /** If true, search is case sensitive. Defaults to false. */
   caseSensitiveSearch: bool,
+  /** Property applied to render data when search result is empty.
+   * Defaults to null
+   * */
```
Comment (helfi92) [inline src/components/MuiTreeView/index.jsx:281]: /** Node to render when `searchTerm` is provided but the search filter returns no result. */

File: src/index.d.ts [hunk 3]
```diff
@@ -62,6 +62,11 @@ export interface MuiTreeViewProps {
    * Defaults to false.
    */
   caseInsensitiveSearch?: boolean;
+  /**
+   * Displays the input JSX element when then search result is empty.
+   * Defaults to null.
+   */
+  onEmptySearch?: any
```
Comment (helfi92) [inline src/index.d.ts:69]: @Slessi Any idea what this should be if one wants to represent the `node` type from https://reactjs.org/docs/typechecking-with-proptypes.html? Thanks!

@ashishzingh missing a semi column at the end.

File: src/components/MuiTreeView/index.jsx [hunk 4]
```diff
@@ -238,6 +245,10 @@ function MuiTreeView(props) {
 
   const graph = createFilteredTree(tree, searchTerm, softSearch);
 
+  if (graph.length === 0) {
+    if (onEmptySearch !== null) return onEmptySearch;
+  }
```
Comment (helfi92) [inline src/components/MuiTreeView/index.jsx:250]:
```jsx
if (!graph.length && onEmptySearch) {
  return onEmptySearch;
}
```

File: src/index.d.ts (hunk 3 above)
Comment (Slessi) [reply src/index.d.ts:69]: @helfi92 I believe `React.ReactNode` === `PropTypes.Node`

Comment (helfi92) [reply src/index.d.ts:69]: Perfect. Thanks @Slessi :)

```ts
onEmptySearch?: React.ReactNode;
```

File: src/components/MuiTreeView/index.jsx (hunk 4 above)
Comment (ashishzingh) [reply src/components/MuiTreeView/index.jsx:250]: @helfi92 i have done the changes. i am still getting the above mentioned warning though.

**Comment by ashishzingh** [comment]
should i write a test case in MuiTreeView_test.js?

**Comment by helfi92** [comment]
Released in v4.2.1 🎉

**Comment by helfi92** [comment]
@ashishzingh I filed https://github.com/helfi92/material-ui-treeview/issues/51 to track this. Comment on the issue directly if you're interested.

**Pull request merged**